addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
                         + "</comments>";
         assertThat(payload, is(expectedPayload));
     }
+
+    @Test
+    public void shouldScanOnlyIfRequestContentTypeIsXml() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
```

Comment:
This seems to be used over and over again, It may as well be extracted to a local method, such as `createNanoHandler`. Then just used as `this.nano.addHandler(createHandler());`

Reply:
A local method has been created: createNanoHandler